data/transactions/logic/assembler.go

```diff
@@ -1043,6 +1043,23 @@ func assembleAppParams(ops *OpStream, spec *OpSpec, args []string) error {
 	return nil
 }
 
+func asmTxField(ops *OpStream, spec *OpSpec, args []string) error {
+	if len(args) != 1 {
+		return ops.errorf("%s expects one argument", spec.Name)
+	}
+	fs, ok := txnFieldSpecByName[args[0]]
```

Comment:
I suppose so.  I was thinking of it as an evaluation issue, so you could look at the opTxField for the definitive list of what's allowed in any Teal version.  It's certainly true that the allowed fields at eval() time must never change for a teal version. So at some point we'll either need to do some version checking in that code, or we'll move to a table based design which would make it easier to use the same value to control assembly and evaluation.

Reply:
I'm deferring this to a followon PR. It will certainly be done, as it will allow us to make the opcode field specs automatically if we encode when (and if) fields are available in a table-driven way.